### PR TITLE
Use preferred `{contact.email_greeting_display}` token over deprecated `{contact.email_greeting}` in ALL templates

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Use preferred `{contact.email_greeting_display}` token over deprecated `{contact.email_greeting}` in templates

This was previously done for some templates - see https://github.com/civicrm/civicrm-core/pull/21847/files#diff-542f63ea529ec93eb6f955b6acccf57c369e37c467ac7513969acb147e059eecL23

Before
----------------------------------------
 `{contact.email_greeting_display}` is preferred / advertised (as part of the switch away from query-object based naming to apiv4 based naming - but templates still have  `{contact.email_greeting}`

After
----------------------------------------
The string is swapped out in templates. No pro-active upgrade on existing sites at this stage

Technical Details
----------------------------------------
Tests cover both tokens

Comments
----------------------------------------
I also fixed the Alex demo data to have this & less deprecated fields (only valid v4 params should be passed in - although some location tokens to work through)

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@seamuslee001 this is also part of the greeting performance patches - the token that matches the db field is cached & will only be looked up once - but the other will have to be looked up twice - I'm switching rather than fixed the deprecated one - tests are good

**Comment by eileenmcnaughton** [comment]
@demeritcowboy are you ok to merge this  - we already did this for some templates - see https://github.com/civicrm/civicrm-core/pull/21847/files#diff-542f63ea529ec93eb6f955b6acccf57c369e37c467ac7513969acb147e059eecL23 - it saves a db lookup & improves standardisation

File: Civi/Test/ExampleData/Contact/Alex.php [hunk 1]
```diff
@@ -12,11 +12,17 @@ public function getExamples(): iterable {
 
   public function build(array &$example): void {
     $example['data'] = [
-      'contact_id' => '100',
+      'id' => '100',
```
Comment (eileenmcnaughton) [inline Civi/Test/ExampleData/Contact/Alex.php:15]: standardise example format to be more like the api style the tokens use

**Comment by demeritcowboy** [comment]
Ok. I spot checked.

My git doesn't like the sample newsletter patch - it says there's a mix of tabs and spaces. Even git has an opinion on that.

**Comment by eileenmcnaughton** [comment]
lol - thanks for that

**Pull request merged**